### ISSUE TITLE
Enhanced Sequence Capabilities (Multiple animation tracks per sequence, modifying instance variables in sequences, etc.)

### ISSUE THREAD

**Issue by McKrak** [YoYoGames/GameMaker-Bugs#6077]
### Is your feature request related to a problem?

As much as I love the idea of sequences in general, their capabilities are quite limited from being able to be utilized to their fullest potential.

### Describe the solution you'd like

One of my issues with the current sequence editor is that we can't set our own instance variables to be manipulated by sequences. While the sequence editor gives us plenty of options when it comes to manipulating instances, developers may also want to manipulate other variables as well. For instance, if you are developing a 3D game in Gamemaker, you can add your z-position variable to the sequence for manipulation. Or if you need to set the depth of an instance through a sequence, you can also add a built-in variable as well.

![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/89408346/a4206918-38b6-4573-94c6-34db28401123)

Next, I would like to be able to "replace" instances in a sequence at initialization. 

If you wanted to run a sequence with an object track using an instance that was already created before the sequence was called, you would have to first create the sequence, then replace the instance using sequence_instance_override_object(). There are two problems with this.

1. You are unnecessarily creating an instance from an object track from a sequence that would end up immediately replaced by the instance you wish to replace with.
2. This is a per-object function, not a per-instance. This means if you wanted to replace a specific instance in a struct, you would not be able to without replacing everything. I do realize for loops are a possible workaround, but it doesn't quite work the same when you already have instances of the same object already created and are calling a sequence only to create more of them. 

A preferable way to solve this problem is by letting us access instances in sequences through their track name. Since the track name doesn't have to be the same as the asset it is creating, we could utilize track names as a way of telling which instances are what. As for the first issue with sequence_instance_override_object() above, perhaps a function to set up instances to set to tracks would be good.

Below is an example of what a feature like this could look like, by adding a new optional parameter to layer_sequence_create():
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/89408346/552534ae-bbed-464d-88e5-05577065d606)

Finally, I feel sequences would truly be complete if it were to also support multiple animations. Currently, a sequence is only one animation. However, what if it was able to contain multiple animations? A feature like this could solve many problems.

For instance, if you wanted to use sequences for your UI system, and wanted to create a button that can be clicked. Your button has multiple animations, such as popping up, fading away, being pressed, being released, and whatever else you want it to do. This cannot be done with current sequences, unless if you're willing to make a new sequence with a different animation and the same object (in which case, you're also creating the same object twice now).

With having multiple animations per sequences, a new function could be introduced: layer_sequence_change(sequence, animation). It takes two arguments, the sequence you wish to change, and the animation you wish to play (assuming animations are uniquely identified by strings from the sequence editor). By using this function, you can make a sequence play a different animation, all while keeping the same instances it already created from the initial animation.

I believe Godot has a similar feature for their equivalent of GameMaker's sequences, but I have only dabbled a tiny bit into Godot as a whole. 

### Additional context

The examples I have provided as solutions are just merely examples. If you see a way they can be improved, feel free to leave feedback about this suggestion. This is something I have wanted in GameMaker for a long time, so seeing this idea of mine come to fruition in its fullest potential means a whole lot to me!

**Comment by KormexGit** [comment]
Related to track names, another way I think track names could be used is to allow them to be used directly to retrieve asset ids, such as:
```
var inst = layer_sequence_get_track_id(my_sequence, "inst_track_1");
var spr = layer_sequence_get_track_id(my_sequence, "sprite_track_1");
inst.some_variable = sprite_get_width(spr);
```
Or, maybe they could even be used as variables directly if used inside a sequence moment, which runs in the scope of the sequence struct:
```
//in a sequence moment
var width = sprite_get_width(sprite_track_1);
inst_track_1.some_variable = width;
```

If something like this become possible along with the stuff in the FR itself, sequences would become way more powerful and also much easier to use.

**Comment by ShivaSouza** [comment]
Yeah, being able to at the very least animate variables inside of sequences is something I miss a lot in Game Maker, It's been suggested many, many times before ( https://github.com/YoYoGames/GameMaker-Bugs/issues/4178 ) and for whatever reason, no one seems to really talk about it: if it's possible, if it's not.

**Comment by 5icheng7hang** [comment]
it's almost been two years since this feature request updated. and I'm still waiting for object variable access from sequence. even if there's way to set the sequence moment into a instance syntax scope can work around this. but currently there's just no way to easily change one specific obj's custom property in sequence or trigger some instance function (even if the variables are defined with in the variable definition editor. which seems to be able to accesss from sequence)
